transports/wasm-ext/src/webtransport/error.rs

```diff
@@ -0,0 +1,38 @@
+use libp2p_core::transport::TransportError;
+use wasm_bindgen::{JsCast, JsValue};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Listening on WebTransport is not possible from within a browser")]
+    ListeningNotPossible,
+
+    #[error("Invalid multiaddr: {0}")]
+    InvalidMultiaddr(&'static str),
+
+    #[error("Noise error: {0}")]
+    Noise(#[from] libp2p_noise::Error),
+
+    #[error("JavaScript error: {0}")]
+    JsError(String),
+}
+
+impl From<JsValue> for Error {
+    fn from(value: JsValue) -> Self {
+        let s = if value.is_instance_of::<js_sys::Error>() {
+            js_sys::Error::from(value)
+                .to_string()
+                .as_string()
+                .unwrap_or_else(|| "Unknown error".to_string())
+        } else {
+            "Unknown error".to_string()
+        };
+
+        Error::JsError(s)
+    }
+}
```

Comment:
This leaks the `wasm-bindgen` version into our public API.

I'd prefer a regular function that we call in `map_err`.